### ISSUE TITLE
dtc: define list of acceptable warnings (and silent them with --warning -no<warnign-name> option)

### ISSUE THREAD

**Issue by erwango** [zephyrproject-rtos/zephyr#7564]
As discussed in #7013, recent version of dtc checks for a list of syntaxic rules and generate warnings when rules are not applied.

Some of these warnings have been corrected. Some other are not corrected for valid reasons. But then we should provide a way for build logs not to be polluted for these "acceptable deviations".

Here is an example of warning we don't want to solve for a good reason (at least not contested so far).
If we do this:
```
green_led_1: led@0 {
		gpios = <&gpiod 2 GPIO_INT_ACTIVE_HIGH>;
		label = "USR1 LED";
};
```
We get this:
```
Warning (unit_address_vs_reg): Node /soc/pin-controller@40020000/usart1@0 has a unit name, but no reg property
```
According to dtc rules, using `: led@0` requires that node provides a `reg` property, this is why this "unit_address_vs_reg" warning is generated.
We have 2 options to solve it:

- Add a `reg = <0>;`. This is possible but does not add specific information: there is no reason for this line in term of HW description
- Remove ` : led@0`. If we do this we loose the ability of using green_led_1 as a phandle and we should use absolute path instead. For instance alias definition `led0 = &green_led_1;` should be replaced by led0 = "/leds/green_led_1". Other example, `	pinctrl-0 = <&usart1_pins_b>;` should be replaced by `pinctrl-0 = "/soc/pin-controller@4200000/usart2_pins_d"; `. Language becomes more heavy to use and hw description less clear.

To sum up, to clear this warning, we have option between adding a useless line or make description less clear.

Since this warning does not do harm, last option is to discard this warning.
dtc provides a way to do this: `--warning no-unit_address_vs_reg` (use "--warning no-\<warning-name\>" for each warning)

Proposal is to add this option in dtc command line in dts.cmake.

**Comment by erwango** [comment]
@galak , should we keep this one ?

**Comment by erwango** [comment]
@galak ping

**Comment by erwango** [comment]
Let me close this one as no need showed up in last year period
